actions/release-candidate/dist/main/index.js

```diff
@@ -31908,27 +31921,38 @@ async function run() {
 			// triggered from a tag, so we fetch it manually so we can verify its tag
 			await exec("git", ["fetch", "origin", "--deepen=1", `+${ github.context.ref }:${ github.context.ref }`]);
 
-			// make sure the tag is signed by a committer in the KEYS file, this
-			// command fails if the tag does not verify.
-			await exec("git", ["tag", "--verify", release_version]);
+			if (do_publish) {
+				// if publishing, tags must be signed with a committers key, download and import committer
+				// keys for verification
+				let committer_keys = "";
+				await exec("curl", [`https://downloads.apache.org/${tlp_dir}/KEYS`], {
+					silent: true,
+					listeners: {
+						stdout: (data) => {
+							committer_keys += data.toString();
+						}
+					}
+				});
+				await exec("gpg", ["--batch", "--import"], {
+					input: Buffer.from(committer_keys)
+				});
+
+				// make sure the tag is signed by a committer in the KEYS file, this
+				// command fails if the tag does not verify.
+				await exec("git", ["tag", "--verify", release_version]);
+			}
 		} else {
-			// this was not triggered by a tag, maybe is was manually triggered via
-			// workflow_dispatch or a normal commit. We should only publish from tags,
-			// so we disable publishing. We also set the release_version so that it has the
+			// this was not triggered by a tag, maybe it was manually triggered via
+			// workflow_dispatch or a normal commit. We also set the release_version so that it has the
 			// same format as a tag (e.g. v1.2.3-rc1)
 			core.warning("Action not triggered from tag, publishing disabled");
 			release_version = `v${ project_version }-rc0`;
-			publish = false;
 		}
 
-		const is_snapshot = project_version.includes("-SNAPSHOT");
-
-		// disable publishing for snapshot builds or non-ASF builds. Note that
-		// publishing could still be disabled if the publish input was explicitly set
-		// to false
-		if (publish && (is_snapshot || process.env.GITHUB_REPOSITORY_OWNER != "apache")) {
+		// warn if publish was explicitly enabled but the version is a snapshot or
+		// not from the ASF
+		if (publish && (is_snapshot || !is_apache)) {
 			core.warning("Publishing disabled for snapshot versions and from non-apache repositories");
-			publish = false;
 		}
```

Comment:
I wonder if we should import this warning. Maybe move it up to after we calculate do_publish and then do something like

```ts
if (!do_publish) {
  core.warning("Publishing disabled:")
  if (!publish) core.warning("- Can only publish if the 'publish' setting is true")
  if (!is_tagged) core.warning("- Can only publish from tagged releases")
  if (is_snapshot) core.warning("- Can only publish non-snapshot versions")
  if (!is_apache) core.warning("- Can only publish from official ASF repositories")  
}
```
I'm not sure I like the wording of those, but something like that so we can make it clear exactly why publishing was disabled.